allensdk/test_utilities/custom_comparators.py

```diff
@@ -0,0 +1,25 @@
+import re
+from typing import Union
+import difflib
+
+
+class WhitespaceStrippedString(object):
+    """Comparator class to compare strings that have been stripped of
+    whitespace. By default removes any unicode whitespace character that
+    matches the regex \s, (which includes [ \t\n\r\f\v], and other unicode
+    whitespace characters).
+    """
+    def __init__(self, string: str, whitespace_chars: str = r"\s",
+                 ASCII: bool = False):
+        self.orig = string
+        self.whitespace_chars = whitespace_chars
+        self.flags = re.ASCII if ASCII else 0
+        self.differ = difflib.Differ()
+        self.value = re.sub(self.whitespace_chars, "", string, self.flags)
+
+    def __eq__(self, other: Union[str, "WhitespaceStrippedString"]):
+        if isinstance(other, str):
+            other = WhitespaceStrippedString(
+                other, self.whitespace_chars, self.flags)
+        self.diff = list(self.differ.compare(self.value, other.value))
```

Comment:
Maybe move the diff to pytest_assertrepr_compare so the object doesn't have a diff relative to something unknown?

Reply:
Context for why this class exists:
pytest does not have good support for custom error messages and comparisons. The reason this class exists is because all custom error messages have to be run through the `pytest_assertrepr_compare` hook which is related to a failing test. In order to detect which custom message should be provided, we have to match based on the instances being compared. Since this is a comparator class meant to be used for a single assertion, I don't think that having the diff on the object is problematic. More importantly though it reduces the lines of code in `pytest_assertrepr_compare`, which has to contain every custom assertion in your codebase. So it could grow unwieldy.

I'd be open to changing the name of the class to make it's functionality clearer though. Do you have any suggestions?